programs/switchDetection2D/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+if(NOT OpenCV_FOUND AND (NOT DEFINED ENABLE_switchDetection2D OR ENABLE_switchDetection2D))
+    message(WARNING "OpenCV package not found, disabling switchDetection2D program")
+endif()
+
+if(NOT TensorflowCC_FOUND AND (NOT DEFINED ENABLE_switchDetection2D OR ENABLE_switchDetection2D))
+    message(WARNING "TensorflowCC package not found, disabling switchDetection2D program")
+endif()
+
+
+cmake_dependent_option(ENABLE_switchDetection2D "Enable/disable switchDetection2D program" ON
+"ENABLE_TravisLib;OpenCV_FOUND;TensorflowCC_FOUND" OFF)
```

Comment:
It is my understanding that this app requires all three components listed on these lines. I'm especially concerned about the Tensorflow dependency. I don't need it in order to use the *haarDetection2D* program, but it is a blocker for me if *switchDetection2D* is meant to serve as a full replacement. Thus, all Tensorflow-related source files in this targets need to be enabled conditionally on the CMake level (via `if` clauses), and macro guards be added to the cpp in which selection is performed on runtime between the three embedded detectors.

On a side note, I guess the *colorRegionDetection2D* and *tensorflowDetection2D* apps will disappear (they're still there on current source directory, whereas *haarDetection2D* has been already replaced).

Reply:
Marked PR as WIP. I expect (or at least hope :blush:) work on what I commented at https://github.com/roboticslab-uc3m/vision/pull/98#issuecomment-558532769 will enable enough polymorphism to be able to load components that are compiled as separate target libraries with clearly differentiated dependencies.

Reply:
Embedding is also an option, though :) (https://github.com/roboticslab-uc3m/vision/issues/79#issuecomment-552521792).